src/peer_manager.rs

```diff
@@ -157,7 +157,12 @@ pub enum PeerState {
     /// We are connected - via a tunnel if the field is `true`.
     Connected(bool),
     /// We are the proxy for the client
-    Client(IpAddr),
+    Client {
+        /// Client IP
+        ip: IpAddr,
+        /// Traffic relayed for Client
```

Comment:
it's actually the `charge` that rate_limiter charges, for get requests, client will be over charged (even consider the traffic of response)
for put requests, the response traffic is not counted in.

maybe the comment can be modified to make it clear?